code/web/src/modules/user/api/actions.js

```diff
@@ -1,145 +1,175 @@
 // Imports
-import axios from 'axios'
-import { query, mutation } from 'gql-query-builder'
-import cookie from 'js-cookie'
+import axios from "axios";
+import { query, mutation } from "gql-query-builder";
+import cookie from "js-cookie";
 
 // App Imports
-import { routeApi } from '../../../setup/routes'
+import { routeApi } from "../../../setup/routes";
 
 // Actions Types
-export const LOGIN_REQUEST = 'AUTH/LOGIN_REQUEST'
-export const LOGIN_RESPONSE = 'AUTH/LOGIN_RESPONSE'
-export const SET_USER = 'AUTH/SET_USER'
-export const LOGOUT = 'AUTH/LOGOUT'
-export const CHANGE_EDIT_MODE = 'AUTH/CHANGE_EDIT_MODE'
-export const SAVE_PROFILE = 'AUTH/SAVE_PROFILE'
+export const LOGIN_REQUEST = "AUTH/LOGIN_REQUEST";
+export const LOGIN_RESPONSE = "AUTH/LOGIN_RESPONSE";
+export const SET_USER = "AUTH/SET_USER";
+export const LOGOUT = "AUTH/LOGOUT";
+export const CHANGE_EDIT_MODE = "AUTH/CHANGE_EDIT_MODE";
+export const SAVE_PROFILE = "AUTH/SAVE_PROFILE";
 
 // Actions
 
 //Set edit mode for user
 
 export function changeEditMode(user) {
-  return dispatch => {
+  return (dispatch) => {
     dispatch({
       type: CHANGE_EDIT_MODE,
-      user
-    })
-  }
+      user,
+    });
+  };
 }
 
 // Set a user after login or using localStorage token
 export function setUser(token, user) {
   if (token) {
-    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+    axios.defaults.headers.common["Authorization"] = `Bearer ${token}`;
   } else {
-    delete axios.defaults.headers.common['Authorization'];
+    delete axios.defaults.headers.common["Authorization"];
   }
 
-  return { type: SET_USER, user }
+  return { type: SET_USER, user };
 }
 
-export function saveProfile(updatedDetails) {
-  return dispatch => {
+export function saveProfile(updatedDetails, subDetails, userSubs) {
```

Comment:
On line 41, `subDetails` is for the `newDate` and `userSubs` is for the array of `id`s.  Maybe we can review the axios post request because I don't quite understand how the keys of the implicitly returned object are being determined.  If we could do that the next time we meet, that would be super helpful!